controllers/packagevariantsets/pkg/controllers/packagevariantset/packagevariantset_controller.go

```diff
@@ -73,6 +73,8 @@ const (
 //+kubebuilder:rbac:groups=config.porch.kpt.dev,resources=packagevariantsets/status,verbs=get;update;patch
 //+kubebuilder:rbac:groups=config.porch.kpt.dev,resources=packagevariantsets/finalizers,verbs=update
 //+kubebuilder:rbac:groups=config.porch.kpt.dev,resources=packagevariants,verbs=create;delete;get;list;patch;update;watch
+//+kubebuilder:rbac:groups=porch.kpt.dev,resources=packagerevisions,verbs=create;delete;get;list;patch;update;watch
+//+kubebuilder:rbac:groups=porch.kpt.dev,resources=packagerevisionresources,verbs=create;delete;get;list;patch;update;watch
```

Comment:
Update controllergen to v0.15.0 and run make generate.
As I mentioned, the setup here is a bit messy. The changes here only get generated if/when we run the "make generate" target. The updated Roles/Bindings are updated but the reality is that the catalog porch pkg itself needs to be updated also. https://github.com/nephio-project/catalog/blob/main/nephio/core/porch/9-porch-controller-packagevariantsets-clusterrole.yaml

Reply:
Hey Fiachra, yes, I have discussed with greg, we're trying to reproduce this issue and re-verify. Might need an update to the code itself. Also, I just re-used the same roles from packagevariant_controller.go. I will also take a look and refactor if this is still needed. I've added the do-not-merge label for now.